File: samples/compiled-dotnet-evidence/csharp/FixtureShapes.cs
using System.Collections.Generic;
using System.Runtime.CompilerServices;

namespace TraceMap.CompiledFixtures.CSharp.Alpha
{
    public interface IExplicit
    {
        string Format(int value);
    }

    public class Widget<T> : IExplicit
    {
        private EventHandler? explicitChanged;
        public event EventHandler? Changed;
        public T? Value;
        public virtual string Name { get; set; } = string.Empty;

        public Widget()
        {
        }

        public Widget(T value) => Value = value;

        public string this[int index]
        {
            get => index.ToString();
            set => _ = value;
        }

        public event EventHandler? ExplicitChanged
        {
            add => explicitChanged += value;
            remove => explicitChanged -= value;
        }

        public int Overload(int value) => value;
        public string Overload(string value) => value;
        public int RefShape(int value) => value;
        public int RefShape(ref int value) => value;
        public int GenericArity<TOne>() => 1;
        public int GenericArity<TOne, TTwo>() => 2;
        public int Optional(int value = 7) => value;
        public TResult Generic<TResult>(TResult value) => value;
        public decimal DecimalShape(decimal value) => value;
        public DateTime DateShape(DateTime value) => value;
        public void TypedReferenceShape(TypedReference value) { }
        public Widget<int>.Nested<string> ConstructedNested(Widget<int>.Nested<string> value) => value;
        public Widget<int>.NonGenericNested ConstructedNonGenericNested(Widget<int>.NonGenericNested value) => value;
        public void Shapes(ref int byRef, out string output, in Guid readOnly, int[] values, int? nullable)
        {
            output = $"{byRef}:{readOnly}:{values.Length}:{nullable}";
        }

        public unsafe int* Pointer(int* value) => value;
        public unsafe delegate* unmanaged[Cdecl]<int> CdeclPointer(delegate* unmanaged[Cdecl]<int> value) => value;
        public unsafe delegate* unmanaged[Stdcall]<int> StdcallPointer(delegate* unmanaged[Stdcall]<int> value) => value;
        string IExplicit.Format(int value) => value.ToString();
        public async Task<int> AsyncShape(int value) { await Task.Yield(); return value; }
        public IEnumerable<int> IteratorShape(int value) { yield return value; }
        public Func<int, int> LambdaShape(int offset) => value => value + offset;
        public void Raise() => Changed?.Invoke(this, EventArgs.Empty);

        public class Nested<TNested>
        {
            public TNested Echo(T left, TNested right) => right;
        }

        public class NonGenericNested
        {
        }
    }

    public sealed class DerivedWidget : Widget<string>
    {
        public override string Name { get; set; } = "derived";
    }

    public sealed class PrimaryConstructorShape(int value)
    {
        public int Value => value;
    }

    public ref struct RefFieldShape
    {
        private ref int value;

        public RefFieldShape(ref int value) => this.value = ref value;
    }
}

namespace TraceMap.CompiledFixtures.CSharp.Beta
{
    public sealed class Widget
    {
        public int Overload(int value) => value + 1;
    }

    public sealed class GlobalNamespaceSentinel
    {
    }
}

File: samples/compiled-dotnet-evidence/vb/FixtureShapes.vb
Namespace TraceMap.CompiledFixtures.VisualBasic
    Public Interface IFormatter
        Function Format(value As Integer) As String
    End Interface

    Public Class BaseWidget
        Public Overridable Function Describe(value As Integer) As String
            Return value.ToString()
        End Function
    End Class

    Public Class Widget
        Inherits BaseWidget
        Implements IFormatter

        Private ReadOnly items(9) As String

        Public Sub New()
        End Sub

        Public Sub New(seed As Integer)
            items(0) = seed.ToString()
        End Sub

        Default Public Property Item(index As Integer) As String
            Get
                Return items(index)
            End Get
            Set(value As String)
                items(index) = value
            End Set
        End Property

        Public ReadOnly Property OptionalItem(Optional index As Integer = 0) As String
            Get
                Return items(index)
            End Get
        End Property

        Public Overloads Function SelectValue(value As Integer) As Integer
            Return value
        End Function

        Public Overloads Function SelectValue(value As String) As String
            Return value
        End Function

        Public Overloads Function GenericArity(Of TOne)() As Integer
            Return 1
        End Function

        Public Overloads Function GenericArity(Of TOne, TTwo)() As Integer
            Return 2
        End Function

        Public Function OptionalValue(Optional value As Integer = 7) As Integer
            Return value
        End Function

        Public Overrides Function Describe(value As Integer) As String
            Return "vb:" & value.ToString()
        End Function

        Public Sub Shapes(ByRef value As Integer, values As Integer())
            value += values.Length
        End Sub

        Private Function Format(value As Integer) As String Implements IFormatter.Format
            Return value.ToString()
        End Function
    End Class

    Public Module ModuleShapes
        Public Function GenericValue(Of T)(value As T) As T
            Return value
        End Function
    End Module
End Namespace

Namespace TraceMap.CompiledFixtures.VisualBasic.Other
    Public Class Widget
        Public Function SelectValue(value As Integer) As Integer
            Return value + 1
        End Function
    End Class
End Namespace
